src/Categories/NaturalTransformation/Monoidal.agda

```diff
@@ -0,0 +1,279 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category.Monoidal
+
+-- Monoidal natural transformations between lax and strong monoidal functors.
+--
+-- See
+--  * John Baez, Some definitions everyone should know,
+--    https://math.ucr.edu/home/baez/qg-fall2004/definitions.pdf
+--  * https://ncatlab.org/nlab/show/monoidal+natural+transformation
+
+module Categories.NaturalTransformation.Monoidal where
+
+open import Level
+open import Data.Product using (_,_)
+
+open import Categories.Category using (module Commutation)
+open import Categories.Category.Product
+import Categories.Category.Monoidal.Reasoning as MonoidalReasoning
+open import Categories.Functor.Monoidal
+open import Categories.Functor.Monoidal.Properties using ()
+  renaming (∘-Monoidal to _∘Fˡ_; ∘-StrongMonoidal to _∘Fˢ_)
+import Categories.Morphism as Morphism
+import Categories.Morphism.Reasoning as MorphismReasoning
+open import Categories.NaturalTransformation as NT using (NaturalTransformation)
+
+open NaturalTransformation renaming (η to _[_])
+
+module Lax where
+
+  module _ {o ℓ e o′ ℓ′ e′}
+           {C : MonoidalCategory o ℓ e} {D : MonoidalCategory o′ ℓ′ e′}
+           (F G : MonoidalFunctor C D) where
+
+    private
+      module C = MonoidalCategory C
+      module D = MonoidalCategory D
+      module F = MonoidalFunctor F renaming (F to U)
+      module G = MonoidalFunctor G renaming (F to U)
+    open D hiding (U; id)
+    open Commutation D.U
+
+    -- Monoidal natural transformations between lax monoidal functors.
+
+    record IsMonoidalNaturalTransformation (β : NaturalTransformation F.U G.U)
+           : Set (o ⊔ ℓ′ ⊔ e′) where
+      field
+        ε-compat      : [ unit ⇒ G.₀ C.unit  ]⟨
+                          F.ε          ⇒⟨ F.₀ C.unit ⟩
+                          β [ C.unit ]
+                        ≈ G.ε
+                        ⟩
+        ⊗-homo-compat : ∀ {X Y} →
+                        [ F.₀ X ⊗₀ F.₀ Y ⇒ G.₀ (X C.⊗₀ Y) ]⟨
+                          F.⊗-homo [ (X , Y) ]  ⇒⟨ F.₀ (X C.⊗₀ Y) ⟩
+                          β [ X C.⊗₀ Y ]
+                        ≈ β [ X ] ⊗₁ β [ Y ]    ⇒⟨ G.₀ X ⊗₀ G.₀ Y ⟩
+                          G.⊗-homo [ (X , Y) ]
+                        ⟩
+
+    record MonoidalNaturalTransformation : Set (o ⊔ ℓ ⊔ ℓ′ ⊔ e′) where
+      field
+        U          : NaturalTransformation F.U G.U
+        isMonoidal : IsMonoidalNaturalTransformation U
+
+      open NaturalTransformation           U          public
+      open IsMonoidalNaturalTransformation isMonoidal public
+
+
+  -- To shorten some definitions
+  private _⇛_ = MonoidalNaturalTransformation
+
+  -- Identity and compositions
+
+  module _ {o ℓ e o′ ℓ′ e′}
+           {C : MonoidalCategory o ℓ e} {D : MonoidalCategory o′ ℓ′ e′} where
+
+    private
+      module C = MonoidalCategory C
```

Comment:
Note that for faster checking (and smaller file sizes), I've been inserting `using` statements on such private module definitions, so that only the needed pieces are copied.

Reply:
What really bugs me: why would that change anything?

Surely, for private definitions, the compiler should be able figure out the dependencies and do this automatically? But I agree that it seems to help sometimes. Is this behavior documented somewhere?

While working on #215 I tried writing a minimal example that exhibits this "bug" (copying terms from a privately re-exported module) but I couldn't see any superfluous terms in the output. So if there's a bug, it doesn't seem to be an obvious one.

So I stopped performing this 'optimization' again. Without being able to trigger the issue explicitly, or finding some documentation of it, it kinda feels like a superstitious ritual...